napalm/nxos_ssh/nxos_ssh.py

```diff
@@ -164,17 +164,21 @@ def parse_intf_section(interface):
 
     if speed_exist:
         match = re.search(re_speed, interface, flags=re.M)
-        speed = float(match.group("speed"))
-        mtu = int(match.group("mtu"))
-        speed_unit = match.group("speed_unit")
-        speed_unit = speed_unit.rstrip(",")
-        # This was alway in Kbit (in the data I saw)
-        if speed_unit != "Kbit":
-            msg = "Unexpected speed unit in show interfaces parsing:\n\n{}".format(
-                interface
-            )
-            raise ValueError(msg)
-        speed = float(speed / 1000.0)
+        if match:
+            speed = float(match.group("speed"))
+            mtu = int(match.group("mtu"))
+            speed_unit = match.group("speed_unit")
+            speed_unit = speed_unit.rstrip(",")
+            # This was alway in Kbit (in the data I saw)
+            if speed_unit != "Kbit":
+                msg = "Unexpected speed unit in show interfaces parsing:\n\n{}".format(
+                    interface
+                )
+                raise ValueError(msg)
+            speed = float(speed / 1000.0)
+        else:
+            speed = -1.0
+            mtu = -1
```

Comment:
For the MTU I agree it should be -1 when this information is missing, but the speed is there, we should just try to match a different regex. Perhaps wrap the MTU section from `re_speed` into a `()?` block? (i.e., `r"(\s+MTU (?P<mtu>\S+)\s+bytes)?,\s+BW\s+(?P<speed>\S+)\s+(?P<speed_unit>\S+).*$"`)